.github/workflows/ci_rust.yml

```diff
@@ -0,0 +1,106 @@
+---
+name: Rust Bindings
+
+on:
+  pull_request:
+    branches:
+      - main
+
+env:
+  # Pin the nightly toolchain to prevent breakage.
+  # This should be occasionally updated.
+  RUST_NIGHTLY_TOOLCHAIN: nightly-2022-04-05
+  MANIFEST_PATH: bindings/rust/Cargo.toml
+
+jobs:
+  generate:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macOS-latest]
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions-rs/toolchain@v1
+        id: toolchain
+        with:
+          toolchain: stable
+          override: true
+
+      - uses: camshaft/rust-cache@v1
+
+      - name: Generate
+        run: ./bindings/rust/generate.sh
```

Comment:
Why run this generate here on both mac and ubuntu ?

Reply:
Generate actually does a bunch of tests as well, so it still needs to be run on both:

https://github.com/aws/s2n-tls/blob/8a01fa5557bda74e8098928592d70eb235524412/bindings/rust/generate.sh#L31-L40

We may want to rename the script to something else? But that should be done separate.